src/tests/test_freq_shift.py

```diff
@@ -0,0 +1,60 @@
+# SPDX-License-Identifier: BSD-3-Clause
```

Comment:
Either shift tests to another class or just called it `test_pulses` or something. You don't usually have a test file for a single feature unless it's amazingly critcal